### PR TITLE
Truncate all long titles inside of plots

### PR DESCRIPTION
**Pull request by sroy3**
Part of #2340 
Screen Shot 2022-09-09 at 10 12 28 AM

We're now parsing the spec and truncating all titles (main title included). 72 lines to truncate titles!

### CODE REVIEW

File: webview/src/plots/components/util.ts [hunk 1]
```diff
@@ -86,3 +87,36 @@ export const truncateTitle = (
   }
   return titleCopy
 }
+
+const isEndValue = (valueType: string) =>
+  ['string', 'number', 'boolean'].includes(valueType)
+
+// eslint-disable-next-line sonarjs/cognitive-complexity
+export const truncateTitles = (spec: Any, size: number, vertical?: boolean) => {
```
[the remark this answers is not in the document]
Comment (sroy3) [reply webview/src/plots/components/util.ts:95]: I blame Vega

**Review by mattseddon — APPROVED** [review]
I think we are already manipulating the spec in the extension. I'll see if I can consolidate the code next week 👍🏻.

**Comment by mattseddon** [comment]
Storybook is showing changes for the table 😕.

**Comment by sroy3** [comment]
> Storybook is showing changes for the table 😕.

Could this be because my branch was not up to date? I updated it, let's see.

**Review by julieg18 — COMMENTED** [review]
Great work!

**Review by julieg18 — APPROVED** [review]

**Pull request merged**